src/library_syscall.js

```diff
@@ -1043,7 +1043,7 @@ function wrapSyscallFunction(x, library, isWasi) {
   pre += "var canWarn = true;\n";
   pre += "var ret = (function() {\n";
   post += "})();\n";
-  post += "if (ret < 0 && canWarn) {\n";
+  post += "if (ret && ret < 0 && canWarn) {\n";
```

Comment:
why does this change?

Reply:
Without this change the `proc_exit` close can see that `proc_exit` never returns (it throws) and it will complain: 
```
building:ERROR: /tmp/emscripten_temp/test_nodefs_rw.js.jso.js:3933:5: WARNING - [JSC_TYPE_MISMATCH] left side of numeric comparison
found   : undefined
required: (bigint|number)
  3933|  if (ret < 0 && canWarn) {
             ^^^
```

Reply:
The test that fails was `core2.test_fs_nodefs_rw_rawfs`.   Without this change that test ends up including the procExit runtime function but after this change `proc_exit` becomes a library syscall function (which get wrapped when SYSCALL_DEBUG is enabled)

Reply:
I see, thanks. Sgtm, and code size doesn't matter as this is in SYSCALL_DEBUG anyhow.

Reply:
I guess we don't have any tests that used the `proc_exit` syscall along with `SYSCALL_DEBUG` today..